.gitignore

```diff
@@ -1,7 +1,17 @@
 *.pyc
 .cache
 .com.apple.timemachine.supported
-bin/
+
+generated/docs/
```

Comment:
Add comment

```
# The "generated" folder is created during a build.
# Normally you wouldn't want those files to be in source control.
# But we actually do keep the latest copy of some of the generated code, specifically the Python modules, for two reasons:
#   1. So that we can make changes to the generator and easily review what effect those changes have in the generated code.
#   2. So that our system tests can create a wheel out of the files and quickly install them, so that they don't need to go through the full code-generation process.
```

(please improve the wording, review for accuracy, and format as needed)

Reply:
Added this comment:
```
# The "generated" folder is created during a build.
# Normally you wouldn't want those files to be in source control.
# But we actually do keep the latest copy of some of the generated code, specifically the Python modules, for two reasons:
#   1. So that we can make changes to the generator and easily review what effect those changes have in the generated code.
#   2. So that our system tests can create a wheel out of the files and quickly install them, instead of going through the full code-generation process.
# These are the generated files we do NOT want it git
```

Reply:
those files -> build artifacts
(small improvement)